### PR TITLE
slicer-macos-preview: Transition build from factory-south-macos to computron

### PR DESCRIPTION
**Pull request by jcfr**
Updates:
* deployment target from 10.13 to 11
* clang from 10.0 to 14.0
* macOS host system from 10.13 (High Sierra) to 13 (Ventura)

Also update `TEST_TO_EXCLUDE_REGEX` to match tests originally specified
for `factory-south-macos`.

See https://github.com/Slicer/Slicer/pull/6695

### CODE REVIEW

File: computron-slicer_preview_nightly.cmake [hunk 1]
```diff
@@ -7,17 +7,17 @@ endmacro()
...
-dashboard_set(HOSTNAME              "computron-macos-tests")
+dashboard_set(HOSTNAME              "computron")
 dashboard_set(OPERATING_SYSTEM      "macOS")
 dashboard_set(SCRIPT_MODE           "Nightly")        # Experimental, Continuous or Nightly
 dashboard_set(Slicer_RELEASE_TYPE   "P")              # (E)xperimental, (P)review or (S)table
 dashboard_set(WITH_PACKAGES         FALSE)             # Enable to generate packages
 dashboard_set(GIT_TAG               "main")         # Specify a tag for Stable release
 if(APPLE)
-  dashboard_set(CMAKE_OSX_DEPLOYMENT_TARGET "10.13")
+  dashboard_set(CMAKE_OSX_DEPLOYMENT_TARGET "11")
```
Comment (jamesobutler) [inline computron-slicer_preview_nightly.cmake:17]: Is it "11" or "11.0"? Or does it not matter?

Comment (jcfr) [reply computron-slicer_preview_nightly.cmake:17]: After the :airplane:  takes off ... I will do some testing and if everything works, I will move forward with integrating this pull-request along with https://github.com/Slicer/Slicer/pull/6695

Comment (jcfr) [reply computron-slicer_preview_nightly.cmake:17]: For sake of consistency with the value specified in https://github.com/Slicer/Slicer/pull/6695, using `11.0` makes sense.

**Comment by jcfr** [comment]
### Update minimum required version

As suggested on the Qt forum[^1], update the minimum deployment target in `qtbase/mkspecs/common/macx.conf` setting `QMAKE_MACOSX_DEPLOYMENT_TARGET` to `11.0` instead of `10.15`

[^1]: https://forum.qt.io/topic/118157/compiling-qt5-13-2-on-macos-catalina/2 

### Attempt to build qtwebengine

Latest version of `qtwebengine` source extracted into Qt source tree:

```
mv qtwebengine qtwebengine.orig
curl -L# https://github.com/qt/qtwebengine/archive/refs/tags/v5.15.12-lts.tar.gz -O
tar -xzvf v5.15.12-lts.tar.gz
mv qtwebengine-5.15.12-lts qtwebengine
```

.. it turns out this was failing with the following warning

```
WARNING: QtWebEngine required submodule qtwebengine-chromium does not exist. Run 'git submodule update --init'.
```

instead source are downloaded using 

```
git clone --recurse-submodules https://github.com/qt/qtwebengine.git -b v5.15.12-lts qtwebengine
```

Despite of this attempt ... the same warning is reported.

**Comment by jamesobutler** [comment]
Ah yes Python 2 for QWebEngine Chromium building…Starting with Qt 6.3 is when Python 3 is used for QWebEngine Chromium builds.

Makes sense to move to 5.15.8 as well to get those latest platform fixes that Slicer users may face. I forgot that the open source download is only available a year after the commercial release.

**Comment by jcfr** [comment]
Attempting to build Qt 5.15.2 ... reports the following error while building the QtWebEngine module:

```
InstalledDir: /Applications/Xcode-14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
Project MESSAGE: Running: /Users/svc-dashboard/D/Support/qt-everywhere-src-5.15.2/qtwebengine/src/3rdparty/gn/out/Release/gn gen /Users/svc-dashboard/D/Support/qt-everywhere-src-5.15.2/qtwebengine/src/core/release --script-executable=/D/Support/qt-everywhere-deps-5.15.2/python-cmake-buildsystem-install/bin/python2 --args='use_qt=true init_stack_vars=false is_component_build=false is_shared=true enable_debugallocation=false enable_media_remoting=false enable_message_center=false enable_nacl=false enable_remoting=false enable_reporting=false enable_resource_whitelist_generation=false enable_swiftshader=false angle_enable_swiftshader=false enable_web_auth=true enable_web_speech=false enable_widevine=true forbid_non_component_debug_builds=false has_native_accessibility=false safe_browsing_mode=0 toolkit_views=false treat_warnings_as_errors=false use_allocator_shim=false use_allocator="none" use_custom_libcxx=false closure_compile=false optimize_webui=false v8_use_external_startup_data=false strip_absolute_paths_from_debug_symbols=false use_jumbo_build=true jumbo_file_merge_limit=8 jumbo_build_excluded=["browser"] enable_precompiled_headers=true is_official_build=true is_debug=false symbol_level=0 blink_symbol_level=0 remove_v8base_debug_symbols=true is_clang=true use_sysroot=false use_system_xcode=true clang_base_path="/Applications/Xcode-14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr" clang_use_chrome_plugins=false mac_deployment_target="10.13" mac_sdk_min="13.1" use_external_popup_menu=false angle_enable_vulkan=false enable_basic_printing=true enable_print_preview=true enable_pdf=true enable_plugins=true enable_spellcheck=true enable_webrtc=true proprietary_codecs=false enable_extensions=true use_kerberos=false have_nodejs=true use_browser_spellchecker=false qtwebengine_target="/Users/svc-dashboard/D/Support/qt-everywhere-src-5.15.2/qtwebengine/src/core/release:QtWebEngineCore"' --root=/Users/svc-dashboard/D/Support/qt-everywhere-src-5.15.2/qtwebengine/src/3rdparty/chromium
ERROR at //build/config/mac/mac_sdk.gni:114:5: Script returned non-zero exit code.
    exec_script("//build/mac/find_sdk.py", find_sdk_args, "list lines")
    ^----------
Current dir: /Users/svc-dashboard/D/Support/qt-everywhere-src-5.15.2/qtwebengine/src/core/release/
Command: /D/Support/qt-everywhere-deps-5.15.2/python-cmake-buildsystem-install/bin/python2 /Users/svc-dashboard/D/Support/qt-everywhere-src-5.15.2/qtwebengine/src/3rdparty/chromium/build/mac/find_sdk.py --print_sdk_path --print_bin_path 13.1
Returned 1.
stderr:

Traceback (most recent call last):
  File "/Users/svc-dashboard/D/Support/qt-everywhere-src-5.15.2/qtwebengine/src/3rdparty/chromium/build/mac/find_sdk.py", line 127, in <module>
    print(main())
  File "/Users/svc-dashboard/D/Support/qt-everywhere-src-5.15.2/qtwebengine/src/3rdparty/chromium/build/mac/find_sdk.py", line 96, in main
    raise Exception('No %s+ SDK found' % min_sdk_version)
Exception: No 13.1+ SDK found

See //components/nacl/features.gni:9:3: whence it was imported.
  import("//build/config/mac/mac_sdk.gni")
  ^--------------------------------------
See //BUILD.gn:18:1: whence it was imported.
import("//components/nacl/features.gni")
^--------------------------------------
Project ERROR: GN run error!
make[3]: *** [sub-gn_run-pro-make_first] Error 3
make[2]: *** [sub-core-make_first] Error 2
make[1]: *** [sub-src-make_first] Error 2
make: *** [module-qtwebengine-make_first] Error 2
```

**Comment by jcfr** [comment]
Error is explained by the following implementation found in the c associated with `v5.15.2`[^1]

https://github.com/qt/qtwebengine-chromium/blob/fb6ab5e483876298235be1c6a6013b426c82b759/chromium/build/mac/find_sdk.py#L91-L96

[^1]: https://github.com/qt/qtwebengine/tree/v5.15.2/src

**Comment by jcfr** [comment]
> Makes sense to move to 5.15.8 as well to get those latest platform fixes that Slicer users may face

It turns out that seems to be addressed in the `chromium` sources associated with `5.15.8`[^1]

| image |
|--|
| https://github.com/qt/qtwebengine-chromium/blob/0ad2814/chromium/build/mac/find_sdk.py#L76-L81 |

[^1]: https://github.com/qt/qtwebengine/tree/5.15.8/src

**Comment by jcfr** [comment]
Attempt to build branch `5.15.8`[^1] leads to the following error:

[^1]: https://github.com/jcfr/qt-easy-build/commits/5.15.8

```
ninja: build stopped: subcommand failed.
make[4]: *** [run_ninja] Error 1  
make[3]: *** [sub-gn_run-pro-make_first] Error 2
make[2]: *** [sub-core-make_first] Error 2
make[1]: *** [sub-src-make_first] Error 2
make: *** [module-qtwebengine-make_first] Error 2
```

Next steps:
* gather more details related to the build error
* attempt to to build qtwebengine `5.15.12` [^2] without the 5.15.8 source tree

[^2]: https://github.com/qt/qtwebengine/tree/5.15.12

**Comment by jamesobutler** [comment]
> attempt to to build qtwebengine 5.15.12 without the 5.15.8 source tree

The below article appears to have good instructions for building latest WebEngine for older Qt versions. Where we could check out https://github.com/qt/qt5/tree/v5.15.8-lts-lgpl and do a git submodule update

https://www.qt.io/blog/building-qt-webengine-against-other-qt-versions

> The easiest way build is the same way the developers of Qt WebEngine does it, as part of a full Qt build from sources. You check out all of Qt from qt5.git, checkout 5.15.2 and do a git submodule update --recursive, then enter the qtwebengine module directory, checkout 5.15 (or 5.15.3 once branched) there and do another git submodule update in the qtwebengine directory to get the right src/3rdparty version. You can then configure and build all of Qt as normal from the qt5 root (see [Building Qt 5 from Git](https://wiki.qt.io/Building_Qt_5_from_Git) again).

> If you instead already have a Qt5 installation and just want to build Qt WebEngine, you again need the qtwebengine git sources (until 5.15.3 source packages are released), checkout the 5.15 branch as above, and then run the qmake from your installed Qt against the source dir of qtwebengine.

**Comment by jcfr** [comment]
> could check out https://github.com/qt/qt5/tree/v5.15.8-lts-lgpl and do a git submodule update

Ditto. 

I tried doing the following while using the `5.15.2` source tree ... 

```
git clone --recurse-submodules --shallow-submodules https://github.com/qt/qtwebengine.git -b v5.15.12-lts qtwebengine
```

.. then I was having issue re-configuring because I didn't use the `-recheck-all` flag despite of having the prerequisites available (python2 and node), which led me to update to `5.15.8` and using the provided sources.

Now trying to re-configure and build after getting the updated sources.

**Comment by jcfr** [comment]
Build of Qt 5.15.8 with QtWebEngine 5.15.12 was successful :heavy_check_mark: 

That said attempting to configure returns the following error:

```
CMake Warning at /D/Support/qt-everywhere-build-5.15.8/lib/cmake/Qt5WebEngineCore/Qt5WebEngineCoreConfig.cmake:93 (find_package):
  Could not find a configuration file for package "Qt5Quick" that is
  compatible with requested version "5.15.12".

  The following configuration files were considered but not accepted:

    /D/Support/qt-everywhere-build-5.15.8/lib/cmake/Qt5Quick/Qt5QuickConfig.cmake, version: 5.15.8

Call Stack (most recent call first):
  /D/Support/qt-everywhere-build-5.15.8/lib/cmake/Qt5WebEngine/Qt5WebEngineConfig.cmake:93 (find_package)
  /D/Support/qt-everywhere-build-5.15.8/lib/cmake/Qt5/Qt5Config.cmake:28 (find_package)
  CMake/SlicerBlockFindQtAndCheckVersion.cmake:22 (find_package)
  CMake/SlicerBlockFindQtAndCheckVersion.cmake:67 (__SlicerBlockFindQtAndCheckVersion_find_qt)
  CMakeLists.txt:662 (include)
```

As indicated here [^1], next steps is to manually update the relevant `Qt5<module>Config.cmake` files to look for Qt 5.15.8 instead of `5.15.12`.

[^1]: https://www.qt.io/blog/building-qt-webengine-against-other-qt-versions

**Comment by jcfr** [comment]
```
$ cd /D/Support/qt-everywhere-build-5.15.8/lib/cmake

$ fgrep -irnsI -E "5.15.12.+FIND_VERSION_EXACT" .
./Qt5WebEngineCore/Qt5WebEngineCoreConfig.cmake:94:                5.15.12 ${_Qt5WebEngineCore_FIND_VERSION_EXACT}
./Qt5PdfWidgets/Qt5PdfWidgetsConfig.cmake:94:                5.15.12 ${_Qt5PdfWidgets_FIND_VERSION_EXACT}
./Qt5WebEngineWidgets/Qt5WebEngineWidgetsConfig.cmake:94:                5.15.12 ${_Qt5WebEngineWidgets_FIND_VERSION_EXACT}
./Qt5Pdf/Qt5PdfConfig.cmake:94:                5.15.12 ${_Qt5Pdf_FIND_VERSION_EXACT}
./Qt5WebEngine/Qt5WebEngineConfig.cmake:94:                5.15.12 ${_Qt5WebEngine_FIND_VERSION_EXACT}
```

**Comment by jcfr** [comment]
After updating the configuration files, configuration completes

```
"/Users/svc-dashboard/D/Support/CMake-3.22.1.app/Contents/bin/cmake" "-DCMAKE_BUILD_TYPE:STRING=Release" "-DCTEST_USE_LAUNCHERS:BOOL=TRUE" "-GUnix Makefiles" "/D/P/S-0"
[...]
-- Configuring done
-- Generating done
-- Build files have been written to: /D/P/A
```

After the experimental build completes, https://github.com/Slicer/Slicer/pull/6695 and this PR (https://github.com/Slicer/DashboardScripts/pull/53) will be merged.

**Comment by jcfr** [comment]
> After the experimental build completes

Updates:
* After the target failing due to a compiled segfault (see below), the build completed
* That said, all `py_` tests starting the application are failing. See https://slicer.cdash.org/viewTest.php?onlyfailed&buildid=2914777


### compiler segfault

_Copied from https://slicer.cdash.org/viewBuildError.php?buildid=2914776_

Error while building C++ object file `CMakeFiles/dcmect_tests.dir/t_roundtrip.cc.o` in target `dcmect_tests`

**Source file**: /Users/svc-dashboard/D/P/A/DCMTK/dcmect/tests/t_roundtrip.cc

**Command:**<details><summary>/Applications/Xcode-14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++</summary><pre>"/Applications/Xcode-14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++" \\
"-DDCMTK_BUILD_IN_PROGRESS" \\
"-DUSE_NULL_SAFE_OFSTRING" \\
"-D_BSD_COMPAT" \\
"-D_BSD_SOURCE" \\
"-D_OSF_SOURCE" \\
"-D_REENTRANT" \\
"-D_XOPEN_SOURCE_EXTENDED" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK-build/config/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/ofstd/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/oflog/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmdata/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmimgle/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmimage/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmjpeg/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmjpls/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmtls/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmnet/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmsr/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmsign/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmwlm/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmqrdb/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmpstat/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmrt/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmiod/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmfg/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmseg/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmtract/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmpmap/include" \\
"-I/Users/svc-dashboard/D/P/A/DCMTK/dcmect/include" \\
"-fvisibility=hidden" \\
"-D_XOPEN_SOURCE_EXTENDED" \\
"-D_BSD_SOURCE" \\
"-D_BSD_COMPAT" \\
"-D_OSF_SOURCE" \\
"-D_DARWIN_C_SOURCE" \\
"-fPIC" \\
"-O3" \\
"-DNDEBUG" \\
"-isysroot" \\
"/Applications/Xcode-14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk" \\
"-mmacosx-version-min=11.0" \\
"-std=c++17" \\
"-MD" \\
"-MT" \\
"dcmect/tests/CMakeFiles/dcmect_tests.dir/t_roundtrip.cc.o" \\
"-MF" \\
"CMakeFiles/dcmect_tests.dir/t_roundtrip.cc.o.d" \\
"-o" \\
"CMakeFiles/dcmect_tests.dir/t_roundtrip.cc.o" \\
"-c" \\
"/Users/svc-dashboard/D/P/A/DCMTK/dcmect/tests/t_roundtrip.cc"
</pre></details>

**Directory:** `/Users/svc-dashboard/D/P/A/DCMTK-build/dcmect/tests`

**Exit Condition:** `254`

**Standard Error:**

```
clang: error: unable to execute command: Segmentation fault: 11
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin22.2.0
Thread model: posix
InstalledDir: /Applications/Xcode-14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
[CTest: warning matched] clang: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
[CTest: warning matched] clang: note: diagnostic msg: /var/folders/sh/x1zm1krn56j808c00cbtr2900000gp/T/t_roundtrip-201b90.cpp
[CTest: warning matched] clang: note: diagnostic msg: /var/folders/sh/x1zm1krn56j808c00cbtr2900000gp/T/t_roundtrip-201b90.sh
[CTest: warning matched] clang: note: diagnostic msg: Crash backtrace is located in
[CTest: warning matched] clang: note: diagnostic msg: /Users/svc-dashboard/Library/Logs/DiagnosticReports/clang_<YYYY-MM-DD-HHMMSS>_<hostname>.crash
[CTest: warning matched] clang: note: diagnostic msg: (choose the .crash file that corresponds to your crash)
[CTest: warning matched] clang: note: diagnostic msg: 

********************
```

**Comment by jcfr** [comment]
Attempting to start Slicer on computron reports the following:

```
$ ./Slicer       
Switch to module:  "Welcome"
Populating font family aliases took 86 ms. Replace uses of missing font family ".AppleSystemUIFont" with one that exists to avoid this cost. 
```

Notes:
* attempting to volume render `MR-Head` works
* existing the application does not report error (exit code 0)
* attempting to run `py_UtilTest` works without reporting any error: Unclear why it failed while running the `Experimental` dashboard

**Comment by jcfr** [comment]
To follow up on this, a build & packaging (without upload to the package server) has been triggered.

See https://slicer.cdash.org/index.php?project=SlicerPreview&date=2023-03-29&filtercount=2&showfilters=1&filtercombine=and&field1=groupname&compare1=63&value1=Nightly&field2=site&compare2=63&value2=computron

**Comment by jcfr** [comment]
>re: compiler segfault

I observed similar failure when building SimpleITK but I was not able to reproduce the crash re-running the helper `.sh` reported in the diagnostic message.

Attempting to observe the load of the system using `htop` while the build was going, memory was still available.

So unlikely to be related to the `-j<N+1> -l<N>` flag introduced in https://github.com/Slicer/DashboardScripts/commit/78f0799118f7d431892162997dcc4f72ecd31228 (based of [this post](https://web.archive.org/web/20210727193205/https://www.preney.ca/paul/archives/341))



<details><summary>Apple clang version 14.0.0 - Error:</summary>

<pre>
clang: error: unable to execute command: Abort trap: 6
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin22.2.0
Thread model: posix
InstalledDir: /Applications/Xcode-14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
[CTest: warning matched] clang: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
[CTest: warning matched] clang: note: diagnostic msg: /var/folders/sh/x1zm1krn56j808c00cbtr2900000gp/T/sitkBinaryDilateImageFilter-225780.cpp
[CTest: warning matched] clang: note: diagnostic msg: /var/folders/sh/x1zm1krn56j808c00cbtr2900000gp/T/sitkBinaryDilateImageFilter-225780.sh
[CTest: warning matched] clang: note: diagnostic msg: Crash backtrace is located in
[CTest: warning matched] clang: note: diagnostic msg: /Users/svc-dashboard/Library/Logs/DiagnosticReports/clang_<YYYY-MM-DD-HHMMSS>_<hostname>.crash
[CTest: warning matched] clang: note: diagnostic msg: (choose the .crash file that corresponds to your crash)
[CTest: warning matched] clang: note: diagnostic msg: 

********************
make[8]: *** [Code/BasicFilters/src/CMakeFiles/SimpleITK_ITKBinaryMathematicalMorphology.dir/sitkBinaryDilateImageFilter.cxx.o] Error 254
make[8]: *** Waiting for unfinished jobs....
make[7]: *** [Code/BasicFilters/src/CMakeFiles/SimpleITK_ITKBinaryMathematicalMorphology.dir/all] Error 2
make[7]: *** Waiting for unfinished jobs....
clang: error: unable to execute command: Segmentation fault: 11
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin22.2.0
Thread model: posix
InstalledDir: /Applications/Xcode-14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
[CTest: warning matched] clang: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
[CTest: warning matched] clang: note: diagnostic msg: /var/folders/sh/x1zm1krn56j808c00cbtr2900000gp/T/sitkCastImageFilter-2v-0136ea.cpp
[CTest: warning matched] clang: note: diagnostic msg: /var/folders/sh/x1zm1krn56j808c00cbtr2900000gp/T/sitkCastImageFilter-2v-0136ea.sh
[CTest: warning matched] clang: note: diagnostic msg: Crash backtrace is located in
[CTest: warning matched] clang: note: diagnostic msg: /Users/svc-dashboard/Library/Logs/DiagnosticReports/clang_<YYYY-MM-DD-HHMMSS>_<hostname>.crash
[CTest: warning matched] clang: note: diagnostic msg: (choose the .crash file that corresponds to your crash)
[CTest: warning matched] clang: note: diagnostic msg: 

********************
make[8]: *** [Code/BasicFilters/src/CMakeFiles/SimpleITK_ITKCommon.dir/sitkCastImageFilter-2v.cxx.o] Error 254
make[8]: *** Waiting for unfinished jobs....
clang: error: unable to execute command: Segmentation fault: 11
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin22.2.0
Thread model: posix
InstalledDir: /Applications/Xcode-14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
[CTest: warning matched] clang: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
[CTest: warning matched] clang: note: diagnostic msg: /var/folders/sh/x1zm1krn56j808c00cbtr2900000gp/T/sitkConnectedComponentImageFilter-786095.cpp
[CTest: warning matched] clang: note: diagnostic msg: /var/folders/sh/x1zm1krn56j808c00cbtr2900000gp/T/sitkConnectedComponentImageFilter-786095.sh
[CTest: warning matched] clang: note: diagnostic msg: Crash backtrace is located in
[CTest: warning matched] clang: note: diagnostic msg: /Users/svc-dashboard/Library/Logs/DiagnosticReports/clang_<YYYY-MM-DD-HHMMSS>_<hostname>.crash
[CTest: warning matched] clang: note: diagnostic msg: (choose the .crash file that corresponds to your crash)
[CTest: warning matched] clang: note: diagnostic msg: 

********************
make[8]: *** [Code/BasicFilters/src/CMakeFiles/SimpleITK_ITKConnectedComponents.dir/sitkConnectedComponentImageFilter.cxx.o] Error 254
make[8]: *** Waiting for unfinished jobs....
make[7]: *** [Code/BasicFilters/src/CMakeFiles/SimpleITK_ITKCommon.dir/all] Error 2
make[7]: *** [Code/BasicFilters/src/CMakeFiles/SimpleITK_ITKConnectedComponents.dir/all] Error 2
clang: error: unable to execute command: Segmentation fault: 11
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin22.2.0
Thread model: posix
InstalledDir: /Applications/Xcode-14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
[CTest: warning matched] clang: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
[CTest: warning matched] clang: note: diagnostic msg: /var/folders/sh/x1zm1krn56j808c00cbtr2900000gp/T/sitkFFTNormalizedCorrelationImageFilter-d93517.cpp
[CTest: warning matched] clang: note: diagnostic msg: /var/folders/sh/x1zm1krn56j808c00cbtr2900000gp/T/sitkFFTNormalizedCorrelationImageFilter-d93517.sh
[CTest: warning matched] clang: note: diagnostic msg: Crash backtrace is located in
[CTest: warning matched] clang: note: diagnostic msg: /Users/svc-dashboard/Library/Logs/DiagnosticReports/clang_<YYYY-MM-DD-HHMMSS>_<hostname>.crash
[CTest: warning matched] clang: note: diagnostic msg: (choose the .crash file that corresponds to your crash)
[CTest: warning matched] clang: note: diagnostic msg: 

********************
make[8]: *** [Code/BasicFilters/src/CMakeFiles/SimpleITK_ITKConvolution.dir/sitkFFTNormalizedCorrelationImageFilter.cxx.o] Error 254
make[8]: *** Waiting for unfinished jobs....
make[7]: *** [Code/BasicFilters/src/CMakeFiles/SimpleITK_ITKConvolution.dir/all] Error 2
make[6]: *** [all] Error 2
make[5]: *** [SimpleITK-prefix/src/SimpleITK-stamp/SimpleITK-build] Error 2
make[4]: *** [CMakeFiles/SimpleITK-build.dir/all] Error 2
make[3]: *** [all] Error 2
</pre>

</details>


References:
* https://developer.apple.com/forums/thread/717259

**Comment by jcfr** [comment]
As a desperate measure ...  I re-triggered a build with `-j9 -l7`

**Comment by jcfr** [comment]
It turns out reducing the load averaged may have helped:

| image |
|--|
| https://slicer.cdash.org/index.php?project=SlicerPreview&date=2023-03-29&filtercount=2&showfilters=1&filtercombine=and&field1=site&compare1=63&value1=computron&field2=groupname&compare2=63&value2=Nightly |

with the following test failures:

image

**Comment by sjh26** [comment]
package works for me 👍

**Comment by jamesobutler** [comment]
Yes I was able to successfully open the package on a Mid-2014 MacBook Pro running macOS Big Sur 11.7.1.

**Comment by pieper** [comment]
Worked for me too - 2019 mac pro with 13.2.1

**Comment by jcfr** [comment]
Thanks for taking the time :pray: This is great news :tada: 

The corresponding changes have been merged into Slicer:
* https://github.com/Slicer/Slicer/pull/6695

We will update the documentation (`Getting Started` & `Prerequisites` sections) in a follow-up pull-request.

**Pull request merged**